scrapers/in/events.py

```diff
@@ -1,90 +1,97 @@
+from .utils import add_space
+
+log = logging.getLogger(__name__)
 
 
 class INEventScraper(Scraper):
     _tz = pytz.timezone("America/Indianapolis")
     # avoid cloudflare blocks for no UA
     cf_headers = {
-        "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36"
+        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) "
+        "Chrome/108.0.0.0 Safari/537.36"  # noqa
     }
-    scraper = cloudscraper.create_scraper()
+    base_url = "https://api.iga.in.gov"
+    session = date.today().year
+
+    def in_request(self, url):
+        """
+        Make request to INDIANA API
+        """
+        apikey = os.environ["INDIANA_API_KEY"]
+        useragent = os.getenv("USER_AGENT", self.cf_headers["User-Agent"])
+        headers = {
+            "Authorization": apikey,
+            "Accept": "application/json",
+            "User-Agent": useragent,
+        }
+        res = requests.get(url, headers=headers)
+
+        if res.status_code != 200:
+            res.raise_for_status()
+        return res
 
     def scrape(self):
-        list_url = "http://iga.in.gov/legislative/{}/committees/standing".format(
-            date.today().year
-        )
-        page = self.scraper.get(list_url).content
-        page = lxml.html.fromstring(page)
-        page.make_links_absolute(list_url)
-
-        for com_row in page.xpath('//li[contains(@class,"committee-item")]/a/@href'):
-            yield from self.scrape_committee_page(com_row)
+        res = self.in_request(f"{self.base_url}/{self.session}/standing-committees")
 
-    def scrape_committee_page(self, url):
-        page = self.scraper.get(url, headers=self.cf_headers).content
-        page = lxml.html.fromstring(page)
-        page.make_links_absolute(url)
-        com = page.xpath('//div[contains(@class, "pull-left span8")]/h1/text()')[
-            0
-        ].strip()
+        for committee in res.json()["items"]:
+            committee_path = committee["link"].replace(
+                "standing-committees", "committees"
+            )
+            url = f"{self.base_url}{committee_path}/meetings"
+            yield from self.extract_committee_events(url)
 
-        for row in page.xpath('//div[contains(@id, "agenda-item")]'):
-            meta = row.xpath('div[contains(@class,"accordion-heading-agenda")]/a')[0]
+    def extract_committee_events(self, url):
 
-            date = meta.xpath("text()")[0].strip()
+        res = self.in_request(url)
+        for meeting in res.json()["items"]:
+            link = meeting["link"]
+            _id = link.split("/")[-1]
 
-            time_and_loc = meta.xpath("span/text()")[0].strip()
-            time_and_loc = time_and_loc.split("\n")
-            time = time_and_loc[0]
-            loc = time_and_loc[1]
+            extra_details = self.in_request(f"{self.base_url}{link}").json()
 
-            if loc == "":
```

Comment:
You may want to hang on to this if statement (after you define the location) as occasionally jurisdictions will create meetings without locations.

Reply:
Understood, this is fixed. Added one more possible option to get the location